BASE/src/Microsoft.ApplicationInsights/TelemetryClient.cs

```diff
@@ -674,6 +676,43 @@ public void Flush()
             }
         }
 
+        /// <summary>
+        /// Asynchronously Flushes the in-memory buffer and any metrics being pre-aggregated.
+        /// </summary>
+        /// <returns>
+        /// True indicates telemetry data ownership is transferred out of process, that are emitted before the flush invocation.
+        /// False indicates transfer of telemetry data has failed, the process still owns all or part of the telemetry.
+        /// </returns>
+        /// <remarks>
+        /// <a href="https://go.microsoft.com/fwlink/?linkid=525722#flushing-data">Learn more</a>
+        /// </remarks>
+        public Task<bool> FlushAsync(CancellationToken cancellationToken)
+        {
+            if (this.TryGetMetricManager(out MetricManager privateMetricManager))
+            {
+                privateMetricManager.Flush(flushDownstreamPipeline: false);
+            }
+
+            TelemetryConfiguration pipeline = this.configuration;
+            if (pipeline != null)
+            {
+                MetricManager sharedMetricManager = pipeline.GetMetricManager(createIfNotExists: false);
+                sharedMetricManager?.Flush(flushDownstreamPipeline: false);
+
+                ITelemetryChannel channel = pipeline.TelemetryChannel;
+
+                if (channel is IAsyncFlushable asyncFlushableChannel && !cancellationToken.IsCancellationRequested)
+                {
+                    return asyncFlushableChannel.FlushAsync(cancellationToken); 
+                }
+            }
+#if NET45
+            return cancellationToken.IsCancellationRequested ? Task.Factory.StartNew(() => false, cancellationToken) : Task.FromResult(false);
+#else
+            return cancellationToken.IsCancellationRequested ? Task.FromCanceled<bool>(cancellationToken) : Task.FromResult(false);
```

Comment:
if we had a valid privateMetricManager on line 693 and flushed those out, we would still return false here, but we actually flushed data. Would we want to return true in that case?